clients/src/main/java/org/apache/kafka/common/requests/OffsetFetchResponse.java

```diff
@@ -39,47 +44,53 @@
     private static final String PARTITION_KEY_NAME = "partition";
     private static final String COMMIT_OFFSET_KEY_NAME = "offset";
     private static final String METADATA_KEY_NAME = "metadata";
-    private static final String ERROR_CODE_KEY_NAME = "error_code";
 
     public static final long INVALID_OFFSET = -1L;
     public static final String NO_METADATA = "";
 
     /**
-     * Possible error codeS:
+     * Possible error codes:
      *
-     *  UNKNOWN_TOPIC_OR_PARTITION (3)
-     *  GROUP_LOAD_IN_PROGRESS (14)
-     *  NOT_COORDINATOR_FOR_GROUP (16)
-     *  TOPIC_AUTHORIZATION_FAILED (29)
-     *  GROUP_AUTHORIZATION_FAILED (30)
+     * - Partition errors:
+     *   - UNKNOWN_TOPIC_OR_PARTITION (3)
+     *   - TOPIC_AUTHORIZATION_FAILED (29)
+     *
+     * - Group or coordinator errors:
+     *   - GROUP_LOAD_IN_PROGRESS (14)
+     *   - NOT_COORDINATOR_FOR_GROUP (16)
+     *   - GROUP_AUTHORIZATION_FAILED (30)
      */
 
+    public static final List<Errors> PARTITION_ERRORS = Arrays.asList(
+            Errors.UNKNOWN_TOPIC_OR_PARTITION,
+            Errors.TOPIC_AUTHORIZATION_FAILED);
+
     private final Map<TopicPartition, PartitionData> responseData;
+    private final Errors error;
+    private Set<Errors> partitionErrors;
```

Comment:
I added this to keep track of partition errors that is needed by `AdminClient` [here](https://github.com/apache/kafka/pull/2074/files/ad48ee99f86cb526c0f7fb23c6e3ce9fd71847a8#diff-66eb7905a403b7fa05ff6bc90e60cd42R125). Unless it's okay to process `PartitionData` on the fly?